llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -2326,7 +2326,7 @@ bool SIInsertWaitcnts::shouldFlushVmCnt(MachineLoop *ML,
         if (MI.mayStore())
           HasVMemStore = true;
       }
-      for (const MachineOperand &Op : MI.all_uses()) {
+      for (const MachineOperand &Op : MI.uses()) {
```

Comment:
Here also, I think `MI` could be anything.

Reply:
Same here, I am excluding VGPR implicit uses, which are never real. Besides the whole function is an opportunistic optimization.